src/netaddress.cpp

```diff
@@ -221,14 +221,18 @@ static void Checksum(Span<const uint8_t> addr_pubkey, uint8_t (&checksum)[CHECKS
 
 }; // namespace torv3
 
-/**
- * Parse a TOR address and set this object to it.
- *
- * @returns Whether or not the operation was successful.
- *
- * @see CNetAddr::IsTor()
- */
-bool CNetAddr::SetSpecial(const std::string& str)
+bool CNetAddr::SetSpecial(const std::string& name)
+{
+    if (SetTor(name)) {
```

Comment:
Validating the string is common to both. I could be moved here maybe.
 
```suggestion
bool CNetAddr::SetSpecial(const std::string& name)
{
    if (!ValidAsCString(str)) {
         return false;
    }
    if (SetTor(name)) {
```

Reply:
Done in https://github.com/vasild/bitcoin/commit/2f493235200a7bc3353b75379ff03a6f729e25c9 (part of [the I2P SAM implementation](https://github.com/vasild/bitcoin/commits/i2p_sam) which I am going to PR soon).